modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/ObjectFieldFormBase.tsx

```diff
@@ -596,45 +596,67 @@ export default function ObjectFieldFormBase({
 												values
 											),
 										});
+
+										if (onSubmit) {
+											onSubmit({
+												...values,
+												listTypeDefinitionExternalReferenceCode:
+													selectedListTypeDefinition.externalReferenceCode,
+												listTypeDefinitionId:
+													selectedListTypeDefinition.id,
+												objectFieldSettings: removeFieldSettings(
+													[
+														'defaultValue',
+														'stateFlow',
+													],
+													values
+												),
+											});
+										}
 									}
+								}}
+								selectedKey={
+									selectedListTypeDefinitionExternalReferenceCode
 								}
-							}}
-							selectedKey={
-								selectedListTypeDefinitionExternalReferenceCode
-							}
-						/>
-
-						<ClayButtonWithIcon
-							aria-label={Liferay.Language.get('refresh-list')}
-							className="lfr-objects__object-field-form-base-picklist-reload-button"
-							data-tooltip-align="top"
-							displayType="secondary"
-							onClick={() =>
-								updateListTypeDefinitions(
-									setListTypeDefinitions
-								)
-							}
-							symbol="reload"
-							title={Liferay.Language.get('refresh-list')}
-						/>
-					</div>
-
-					<ClayButton
-						aria-labelledby={Liferay.Language.get(
-							'manage-picklists'
-						)}
-						className="lfr-objects__object-field-form-base-picklist-manage-button"
-						displayType="secondary"
-						onClick={() => {
-							window.open(listTypeDefinitionsURL, '_blank');
-						}}
-					>
-						<span className="icon">
-							{Liferay.Language.get('manage-picklists')}
-						</span>
-
-						<ClayIcon symbol="shortcut" />
-					</ClayButton>
+							/>
+
+							<ClayButtonWithIcon
+								aria-label={Liferay.Language.get(
+									'refresh-list'
+								)}
+								className="lfr-objects__object-field-form-base-picklist-reload-button"
+								data-tooltip-align="top"
+								displayType="secondary"
+								onClick={() =>
+									updateListTypeDefinitions(
+										setListTypeDefinitions
+									)
+								}
+								symbol="reload"
+								title={Liferay.Language.get('refresh-list')}
+							/>
+
+							<ClayButton
+								aria-labelledby={Liferay.Language.get(
+									'manage-picklists'
+								)}
+								className="lfr-objects__object-field-form-base-picklist-manage-button"
+								displayType="secondary"
+								onClick={() => {
+									window.open(
+										listTypeDefinitionsURL,
+										'_blank'
+									);
+								}}
+							>
+								<span className="icon">
+									{Liferay.Language.get('manage-picklists')}
+								</span>
+
+								<ClayIcon symbol="shortcut" />
```

Comment:
Hey @Artartur, reloading all of these two buttons is unnecessary as the only component that needs to be reloaded is the SingleSelect, so you can remove it from the condition, you can also remove the div around the entire inside of the condition, the condition itself will be only, please tell me if you have any question about this:

```tsx
{reload ? (
	<ClayLoadingIndicator
		displayType="secondary"
		size="sm"
	/>
) : (
	<SingleSelect
		className="lfr-objects__object-field-form-base-picklist-select-field"
		disabled={disabled}
		error={errors.listTypeDefinitionId}
		id="objectFieldFormBase"
		items={listTypeDefinitionsItems}
		label={Liferay.Language.get('picklist')}
		onSelectionChange={(value) => {
			const selectedListTypeDefinition = listTypeDefinitions.find(
				({externalReferenceCode}) =>
					externalReferenceCode === value
			);
			if (selectedListTypeDefinition) {
				setValues({
					listTypeDefinitionExternalReferenceCode:
						selectedListTypeDefinition.externalReferenceCode,
					listTypeDefinitionId:
						selectedListTypeDefinition.id,
					objectFieldSettings: removeFieldSettings(
						['defaultValue', 'stateFlow'],
						values
					),
				});

				if (onSubmit) {
					onSubmit({
						...values,
						listTypeDefinitionExternalReferenceCode:
							selectedListTypeDefinition.externalReferenceCode,
						listTypeDefinitionId:
							selectedListTypeDefinition.id,
						objectFieldSettings: removeFieldSettings(
							[
								'defaultValue',
								'stateFlow',
							],
							values
						),
					});
				}
			}
		}}
		selectedKey={
			selectedListTypeDefinitionExternalReferenceCode
		}
	/>
)}
```

Reply:
Done!